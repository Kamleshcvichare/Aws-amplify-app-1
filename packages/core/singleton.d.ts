export * from './lib-esm/Singleton';