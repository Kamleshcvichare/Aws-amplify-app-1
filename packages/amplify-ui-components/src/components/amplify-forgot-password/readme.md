# amplify-forgot-password



<!-- Auto Generated Below -->


## Properties

| Property                | Attribute            | Description | Type                                                | Default                 |
| ----------------------- | -------------------- | ----------- | --------------------------------------------------- | ----------------------- |
| `formFields`            | --                   |             | `FormFieldTypes`                                    | `undefined`             |
| `handleAuthStateChange` | --                   |             | `(nextAuthState: AuthState, data?: object) => void` | `undefined`             |
| `headerText`            | `header-text`        |             | `string`                                            | `'Reset your password'` |
| `overrideStyle`         | `override-style`     |             | `boolean`                                           | `false`                 |
| `submitButtonText`      | `submit-button-text` |             | `string`                                            | `'Send Code'`           |


## Dependencies

### Used by

 - [amplify-authenticator](../amplify-authenticator)

### Depends on

- [amplify-form-section](../amplify-form-section)
- [amplify-link](../amplify-link)
- [amplify-auth-fields](../amplify-auth-fields)

### Graph
```mermaid
graph TD;
  amplify-forgot-password --> amplify-form-section
  amplify-forgot-password --> amplify-link
  amplify-forgot-password --> amplify-auth-fields
  amplify-form-section --> amplify-button
  amplify-form-section --> amplify-section
  amplify-auth-fields --> amplify-username-field
  amplify-auth-fields --> amplify-password-field
  amplify-auth-fields --> amplify-email-field
  amplify-auth-fields --> amplify-code-field
  amplify-auth-fields --> amplify-form-field
  amplify-auth-fields --> amplify-form-field
  amplify-username-field --> amplify-form-field
  amplify-form-field --> amplify-label
  amplify-form-field --> amplify-input
  amplify-form-field --> amplify-hint
  amplify-password-field --> amplify-form-field
  amplify-email-field --> amplify-form-field
  amplify-code-field --> amplify-form-field
  amplify-authenticator --> amplify-forgot-password
  style amplify-forgot-password fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
