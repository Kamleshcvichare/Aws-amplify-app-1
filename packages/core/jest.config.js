module.exports = {
	...require('../../jest.config'),
	coverageThreshold: {
		global: {
			branches: 74, // todo(ashwinkumar6): temp, add tests and bump back
			functions: 72,
			lines: 87, // todo(ashwinkumar6): temp, add tests and bump back
			statements: 88,
		},
	},
};
