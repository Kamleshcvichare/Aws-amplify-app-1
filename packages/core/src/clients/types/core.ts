/**
 * General None HTTP-specific request interface
 */
export interface Request {
	url: URL;
	body?: unknown;
}

export interface Response {
	body: unknown;
}

export interface TransferHandler<
	Input extends Request,
	Output extends Response,
	TransferOptions
> {
	(request: Input, options: TransferOptions): Promise<Output>;
}

/**
 * A slimmed down version of the AWS SDK v3 middleware handler, only handling instantiated requests
 */
export type MiddlewareHandler<
	Input extends Request,
	Output extends Response,
	MiddlewareOptions
> = (request: Input, options: MiddlewareOptions) => Promise<Output>;

export type MiddlewareContext = Record<string, unknown>;

/**
 * A slimmed down version of the AWS SDK v3 middleware, only handling tasks after Serde.
 */
export type Middleware<
	Input extends Request,
	Output extends Response,
	MiddlewareOptions
> = (
	next: MiddlewareHandler<Input, Output, MiddlewareOptions>,
	context: MiddlewareContext
) => MiddlewareHandler<Input, Output, MiddlewareOptions>;
