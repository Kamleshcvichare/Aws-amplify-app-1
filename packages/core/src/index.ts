// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

/*
This file maps top-level exports from `@aws-amplify/core`. These are intended to be potentially customer-facing exports.
*/
// Hub exports
export { Hub } from './Hub';
export { HubCapsule, HubCallback, HubPayload } from './Hub/types';

// Singleton exports
export {
	TokenProvider,
	AuthTokens,
	FetchAuthSessionOptions,
	AWSCredentialsAndIdentityIdProvider,
	AWSCredentialsAndIdentityId,
	Identity,
	OAuthConfig,
	CognitoUserPoolConfig,
} from './singleton/Auth/types';
export {
	AuthConfig,
	AuthUserPoolConfig,
	AuthUserPoolAndIdentityPoolConfig,
	APIConfig,
	CacheConfig,
	StorageAccessLevel,
	StorageConfig,
	GetCredentialsOptions,
	ResourcesConfig,
	LibraryOptions,
	AnalyticsConfig,
	GeoConfig,
} from './singleton/types';
export {
	Amplify,
	fetchAuthSession,
	AmplifyClass as AmplifyClassV6,
	clearCredentials,
} from './singleton';

// AWSClients exports
export {
	getCredentialsForIdentity,
	getId,
	GetCredentialsForIdentityInput,
	GetCredentialsForIdentityOutput,
} from './awsClients/cognitoIdentity';

// Amplify-wide constructs
export { UserProfile } from './types';

// Storage helpers
export {
	CookieStorage,
	defaultStorage,
	sessionStorage,
	sharedInMemoryStorage,
} from './storage';
export { KeyValueStorageInterface } from './types';

// Cache exports
export { Cache } from './Cache';

// Internationalization utilities
export { I18n } from './I18n';

// Logging utilities
export { ConsoleLogger } from './Logger';
