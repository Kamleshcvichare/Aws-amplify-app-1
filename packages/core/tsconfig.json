{
	"extends": "../../tsconfig.json",
	"compilerOptions": {
		"baseUrl": ".",
		"importHelpers": false,
		"paths": {
			"~/*": ["./*"]
		}
	},
	"include": ["./src", "__tests__"]
}
