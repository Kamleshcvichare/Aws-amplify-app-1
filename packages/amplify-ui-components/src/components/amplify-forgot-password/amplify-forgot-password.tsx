import { Component, Prop, State, h } from '@stencil/core';
import { FormFieldTypes, PhoneNumberInterface } from '../amplify-auth-fields/amplify-auth-fields-interface';
import { AuthState, AuthStateHandler, UsernameAttributes } from '../../common/types/auth-types';
import {
  RESET_YOUR_PASSWORD,
  SEND_CODE,
  BACK_TO_SIGN_IN,
  NO_AUTH_MODULE_FOUND,
  COUNTRY_DIAL_CODE_DEFAULT,
  COUNTRY_DIAL_CODE_SUFFIX,
  PHONE_SUFFIX,
} from '../../common/constants';
import { CodeDeliveryType } from './amplify-forgot-password-interface';

import { Auth } from '@aws-amplify/auth';
import { Logger } from '@aws-amplify/core';
import { dispatchToastHubEvent, dispatchAuthStateChangeEvent, composePhoneNumberInput } from '../../common/helpers';

const logger = new Logger('ForgotPassword');

@Component({
  tag: 'amplify-forgot-password',
  shadow: true,
})
export class AmplifyForgotPassword {
  /** The header text of the forgot password section */
  @Prop() headerText: string = RESET_YOUR_PASSWORD;
  /** The text displayed inside of the submit button for the form */
  @Prop() submitButtonText: string = SEND_CODE;
  /** The form fields displayed inside of the forgot password form */
  @Prop() formFields: FormFieldTypes;
  /** The function called when making a request to reset password */
  @Prop() handleSend: (event: Event) => void = event => this.send(event);
  /** The function called when submitting a new password */
  @Prop() handleSubmit: (event: Event) => void = event => this.submit(event);
  /** Passed from the Authenticator component in order to change Authentication state */
  @Prop() handleAuthStateChange: AuthStateHandler = dispatchAuthStateChangeEvent;

  @Prop() usernameAttributes: UsernameAttributes = 'username';

  @State() username: string;
  @State() password: string;
  @State() code: string;
  @State() delivery: CodeDeliveryType | null = null;
  @State() loading: boolean = false;
  @State() email: string;
  @State() phoneNumber: PhoneNumberInterface = {
    countryDialCodeValue: COUNTRY_DIAL_CODE_DEFAULT,
    phoneNumberValue: null,
  };

  componentWillLoad() {
    switch (this.usernameAttributes) {
      case 'email':
        this.formFields = [
          {
            type: 'email',
            required: true,
            handleInputChange: event => this.handleEmailChange(event),
            inputProps: {
              'data-test': 'email-input',
            },
          },
        ];
        break;
      case 'phone_number':
        this.formFields = [
          {
            type: 'phone_number',
            required: true,
            handleInputChange: event => this.handlePhoneNumberChange(event),
            inputProps: {
              'data-test': 'phone-number-input',
            },
          },
        ];
        break;
      case 'username':
      default:
        this.formFields = [
          {
            type: 'username',
            required: true,
            handleInputChange: event => this.handleUsernameChange(event),
            value: this.username,
            inputProps: {
              'data-test': 'username-input',
            },
          },
        ];
        break;
    }
  }

  handleUsernameChange(event) {
    this.username = event.target.value;
  }

  handleEmailChange(event) {
    this.email = event.target.value;
  }

  handlePhoneNumberChange(event) {
    const name = event.target.name;
    const value = event.target.value;

    /** Cognito expects to have a string be passed when signing up. Since the Select input is separate
     * input from the phone number input, we need to first capture both components values and combined
     * them together.
     */

    if (name === COUNTRY_DIAL_CODE_SUFFIX) {
      this.phoneNumber.countryDialCodeValue = value;
    }

    if (name === PHONE_SUFFIX) {
      this.phoneNumber.phoneNumberValue = value;
    }
  }

  handlePasswordChange(event) {
    this.password = event.target.value;
  }

  handleCodeChange(event) {
    this.code = event.target.value;
  }

  async send(event) {
    if (event) {
      event.preventDefault();
    }
    if (!Auth || typeof Auth.forgotPassword !== 'function') {
      throw new Error(NO_AUTH_MODULE_FOUND);
    }
    this.loading = true;

    switch (this.usernameAttributes) {
      case 'email':
        this.username = this.email;
        break;
      case 'phone_number':
        this.username = composePhoneNumberInput(this.phoneNumber);
        break;
      case 'username':
      default:
        this.username = this.username;
        break;
    }

    try {
      const data = await Auth.forgotPassword(this.username);
      logger.debug(data);
      this.formFields = [
        {
          type: 'code',
          required: true,
          handleInputChange: event => this.handleCodeChange(event),
          value: this.code,
        },
        {
          type: 'password',
          required: true,
          handleInputChange: event => this.handlePasswordChange(event),
          label: 'New password',
          placeholder: 'Enter your new password',
          value: this.password,
        },
      ];
      this.delivery = data.CodeDeliveryDetails;
    } catch (error) {
      dispatchToastHubEvent(error);
    } finally {
      this.loading = false;
    }
  }

  async submit(event: Event) {
    if (event) {
      event.preventDefault();
    }
    if (!Auth || typeof Auth.forgotPasswordSubmit !== 'function') {
      throw new Error(NO_AUTH_MODULE_FOUND);
    }
    this.loading = true;
    try {
      const data = await Auth.forgotPasswordSubmit(this.username, this.code, this.password);
      logger.debug(data);
      this.handleAuthStateChange(AuthState.SignIn);
      this.delivery = null;
    } catch (error) {
      dispatchToastHubEvent(error);
    } finally {
      this.loading = false;
    }
  }

  render() {
    const submitFn = this.delivery ? event => this.handleSubmit(event) : event => this.handleSend(event);
    return (
      <amplify-form-section
        headerText={this.headerText}
        handleSubmit={submitFn}
        loading={this.loading}
        secondaryFooterContent={
          <amplify-link
            onClick={() => this.handleAuthStateChange(AuthState.SignIn)}
            data-test="forgot-password-back-to-sign-in-link"
          >
            {BACK_TO_SIGN_IN}
          </amplify-link>
        }
        testDataPrefix={'forgot-password'}
      >
        <amplify-auth-fields formFields={this.formFields} />
      </amplify-form-section>
    );
  }
}
