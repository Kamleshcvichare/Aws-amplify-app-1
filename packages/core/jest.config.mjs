import { getJestConfig } from '../../jest/getJestConfig.mjs';
import tsconfig from './tsconfig.json' assert { type: 'json' };

const jestConfig = getJestConfig(tsconfig.compilerOptions, {
	coverageThreshold: {
		global: {
			branches: 75,
			functions: 72,
			lines: 88,
			statements: 88,
		},
	},
});

export default jestConfig;
