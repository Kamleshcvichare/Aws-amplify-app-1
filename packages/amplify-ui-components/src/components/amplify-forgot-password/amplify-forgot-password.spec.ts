import { newSpecPage } from '@stencil/core/testing';
import { AmplifyForgotPassword } from './amplify-forgot-password';

describe('amplify-forgot-password spec:', () => {
  describe('Component logic ->', () => {
    let amplifyForgotPassword;

    beforeEach(() => {
      amplifyForgotPassword = new AmplifyForgotPassword();
    });

    it('`sceneName` should be undefined by default', () => {
      expect(amplifyForgotPassword.sceneName).toBeUndefined();
    });

    it('`loading` should be false by default', () => {
      expect(amplifyForgotPassword.loading).toBeFalsy();
    });
  });
  describe('Render logic ->', () => {
    it(`should render loading overlay by default`, async () => {
      const page = await newSpecPage({
        components: [AmplifyForgotPassword],
        html: `<amplify-forgot-password sceneName={'scene1'} />`,
      });
      expect(page.root).toMatchSnapshot();
    });
  });
});
