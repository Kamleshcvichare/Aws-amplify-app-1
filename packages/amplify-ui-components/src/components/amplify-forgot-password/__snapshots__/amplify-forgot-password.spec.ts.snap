// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-forgot-password spec: Render logic -> should render loading overlay by default 1`] = `
<amplify-forgot-password scenename="{'scene1'}">
  <amplify-form-section headertext="Reset your password">
    <amplify-auth-fields></amplify-auth-fields>
  </amplify-form-section>
</amplify-forgot-password>
`;
